File: src/sequence/mergeOrderedLists/test_merge_ordered_lists.py
from merge_ordered_lists import *
import pytest

data = [
    ([1, 2, 3], [5, 6, 7, 8], [1, 2, 3, 5, 6, 7, 8]),
    ([1, 7, 10, 16], [5, 6, 13, 20], [1, 5, 6, 7, 10, 13, 16, 20]),
    ([1], [2], [1, 2]),
    ([1, 2], [1, 3], [1, 1, 2, 3]),
    ([10, 11, 12], [1, 2,  3], [1, 2, 3, 10, 11, 12]),
]


@pytest.mark.parametrize('list1, list2, expectedList', data)
def test_merge_ordered_lists1(list1, list2, expectedList):
    assert merge_ordered_lists1(list1, list2) == expectedList


@pytest.mark.parametrize('list1, list2, expectedList', data)
def test_merge_ordered_lists2(list1, list2, expectedList):
    assert merge_ordered_lists2(list1, list2) == expectedList

File: src/sequence/mergeOrderedLists/TestMergeOrderedLists.java
package sequence.mergeOrderedLists;

import static org.junit.Assert.assertEquals;
import java.util.List;
import java.util.stream.Stream;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;
import static sequence.mergeOrderedLists.MergeOrderedLists.*;

public class TestMergeOrderedLists {
    @ParameterizedTest
    @MethodSource("provideParams")
    void testMergeLists(List<Integer> list1,
            List<Integer> list2, List<Integer> expected) {
        List<Integer> actual = mergeOrderedLists(list1, list2);
        assertEquals(expected, actual);
    }

    private static Stream<Arguments> provideParams() {
        return Stream.of(
                Arguments.of(List.of(1, 2, 3),
                        List.of(5, 6, 7, 8),
                        List.of(1, 2, 3, 5, 6, 7, 8)),
                Arguments.of(List.of(1, 7, 10, 16),
                        List.of(5, 6, 13, 20),
                        List.of(1, 5, 6, 7, 10, 13, 16, 20)),
                Arguments.of(List.of(1),
                        List.of(2),
                        List.of(1, 2)),
                Arguments.of(List.of(1, 2),
                        List.of(1, 3),
                        List.of(1, 1, 2, 3)),
                Arguments.of(List.of(10, 11, 12),
                        List.of(1, 2, 3),
                        List.of(1, 2, 3, 10, 11, 12)),
                Arguments.of(List.of(1),
                        List.of(),
                        List.of(1)),
                Arguments.of(List.of(),
                        List.of(),
                        List.of()),
                Arguments.of(List.of(1),
                        List.of(1),
                        List.of(1, 1)));
    }
}
